<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1920</width>
    <height>1000</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>1920</width>
    <height>1080</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>301</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>Serial</string>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
     <property name="sizeIncrement">
      <size>
       <width>10</width>
       <height>10</height>
      </size>
     </property>
     <property name="text">
      <string>Port Name:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="ccb_portname">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>20</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Baudrate:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="ccb_baudrate">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>70</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_open">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>120</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Open</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_portstatus">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>220</y>
      <width>701</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Data Transaction</string>
    </property>
    <widget class="QListWidget" name="list_dataReceive">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>681</width>
       <height>121</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>0</y>
      <width>281</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Motor 1</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>261</width>
       <height>141</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Kp:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Ki:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Kd:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="edt_kd"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Vref:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="edt_vref"/>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="edt_ki"/>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="edt_kp"/>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="btn_sendpidm1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>SET PID </string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_sendvrefm1">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>170</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>SET VREF</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>0</y>
      <width>271</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Motor 2</string>
    </property>
    <widget class="QWidget" name="layoutWidget_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>251</width>
       <height>141</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="2" column="1">
       <widget class="QLineEdit" name="edt_kd_2"/>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="edt_ki_2"/>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Kp:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Ki:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="edt_kp_2"/>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="edt_vref_2"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Kd:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Vref:</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="btn_sendpidm2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>SET PID </string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_sendvrefm2">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>170</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>SET VREF</string>
     </property>
    </widget>
   </widget>
   <widget class="QCustomPlot" name="Motor1_Chart" native="true">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>440</y>
      <width>651</width>
      <height>501</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>611</width>
      <height>0</height>
     </size>
    </property>
   </widget>
   <widget class="QCustomPlot" name="Motor2_Chart" native="true">
    <property name="geometry">
     <rect>
      <x>1270</x>
      <y>440</y>
      <width>611</width>
      <height>501</height>
     </rect>
    </property>
   </widget>
   <widget class="QCustomPlot" name="Map_Chart" native="true">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>10</y>
      <width>971</width>
      <height>421</height>
     </rect>
    </property>
   </widget>
   <widget class="QGroupBox" name="grb_Stanley">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>500</y>
      <width>251</width>
      <height>201</height>
     </rect>
    </property>
    <property name="title">
     <string>Stanley Control</string>
    </property>
    <widget class="QLineEdit" name="edt_Kgain">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>30</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>12</x>
       <y>31</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>K Gain :</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="edt_Ksoft">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>80</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>K Soft :</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_AvoidFlag">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>120</y>
       <width>20</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>121</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Avoiding Flag:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_SetStanley">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>150</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>SET STANLEY</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grb_Fuzzy">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>500</y>
      <width>261</width>
      <height>201</height>
     </rect>
    </property>
    <property name="title">
     <string>Fuzzy Control</string>
    </property>
    <widget class="QLabel" name="label_24">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Ke :</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="edt_Ke">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>30</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_25">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Ke_dot :</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="edt_Ke_dot">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>70</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="edt_Vmax">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>110</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_26">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Vmax:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_SetFuzzy">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>150</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>SET FUZZY</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grb_VehicleInfo">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>710</y>
      <width>591</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>Vehicle Info</string>
    </property>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Ref angle: (rad)</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Current angle:(rad) </string>
     </property>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>V linear:(m/s) </string>
     </property>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>40</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>V1 Ref:(m/s)</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_18">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>160</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>V2 Ref:(m/s)</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_19">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>80</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>V1 Current:(m/s)</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_20">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>120</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>V2 Current:(m/s)</string>
     </property>
    </widget>
    <widget class="QLabel" name="labl_RefAngle">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>40</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="labl_CurrentAngle">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>80</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="labl_VLinear">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>120</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="labl_V1Ref">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>40</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="labl_V1Current">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>80</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="labl_V2Ref">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>160</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="labl_V2Current">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>120</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_21">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>160</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Current Point:</string>
     </property>
    </widget>
    <widget class="QLabel" name="labl_CurrentPoint">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>160</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_23">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Total Points:</string>
     </property>
    </widget>
    <widget class="QLabel" name="labl_TotalPoint">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>200</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>230</y>
      <width>161</width>
      <height>191</height>
     </rect>
    </property>
    <property name="title">
     <string>Vehicle control</string>
    </property>
    <widget class="QPushButton" name="btn_SendData">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Send Data</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_stop">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Start</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_cleardata">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>380</y>
      <width>551</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Map setting</string>
    </property>
    <widget class="QPushButton" name="btn_readMap">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Read Map</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_drawMap">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>50</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Draw map</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_SendMap">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>50</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Send Map</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_ClearMap">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>50</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear Map</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_22">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>400</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1920</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>ccb_portname</tabstop>
  <tabstop>ccb_baudrate</tabstop>
  <tabstop>btn_open</tabstop>
  <tabstop>edt_kp</tabstop>
  <tabstop>edt_ki</tabstop>
  <tabstop>edt_kd</tabstop>
  <tabstop>edt_vref</tabstop>
  <tabstop>edt_kp_2</tabstop>
  <tabstop>edt_ki_2</tabstop>
  <tabstop>edt_kd_2</tabstop>
  <tabstop>edt_vref_2</tabstop>
  <tabstop>btn_sendpidm1</tabstop>
  <tabstop>btn_sendvrefm1</tabstop>
  <tabstop>list_dataReceive</tabstop>
  <tabstop>btn_sendpidm2</tabstop>
  <tabstop>btn_sendvrefm2</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
